src/HealthGPS.Datastore/zip_file.cpp

```diff
@@ -0,0 +1,55 @@
+#include "zip_file.h"
+
+#include <fmt/format.h>
+#include <libzippp.h>
+
+#include <fstream>
+#include <random>
+
+namespace hgps::data {
+std::filesystem::path create_temporary_directory() {
+    auto tmp_dir = std::filesystem::temp_directory_path();
```

Comment:
I'm wondering how well this behaves on e.g. HPC: is a large array job going to fill up `/tmp` with multiple identical extractions, or does this provide some cleanup in the end?

Reply:
If it's an issue, perhaps we should extract the zip to the same directory, after checking it doesn't exist or is of bad integrity?

Reply:
I haven't written any cleanup code myself, but `/tmp` should usually be periodically cleaned by the system so I don't think it will be an issue. In any case, it's only a few megabytes' worth of data that we're extracting anyway (at least for now).

Reply:
Okay. Maybe we should open an issue to remind ourselves to add cleanup in future, if it becomes an issue.

Reply:
Yeah, I was a bit worried about the clobbering thing, which is why I wanted to use a unique folder for each extraction.

I'm planning on adding caching for downloaded zip files at some point, so could also use the cache folder for extracing them to? On Linux this would be something like `.cache/healthgps`. We'd have to be a little careful to make sure there aren't race conditions with extracting and reading the zip files (e.g. for array jobs), but this should be doable. What do you think?

Reply:
Sounds good. :+1:  So long as those race conditions can be avoided!

Reply:
I think the trick is to create a temporary folder with a unique name in the same folder and then rename it once the extraction is complete.

Reply:
#400